solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -1400,37 +1540,13 @@ public String getBaseUrlForNodeName(final String nodeName) {
   /** Watches a single collection's state.json. */
   class StateWatcher implements Watcher {
     private final String coll;
-    private final String collectionPath;
 
     StateWatcher(String coll) {
       this.coll = coll;
-      collectionPath = DocCollection.getCollectionPath(coll);
     }
 
     @Override
-    public void process(WatchedEvent event) {
-      // session events are not change events, and do not remove the watcher
-      if (EventType.None.equals(event.getType())) {
-        return;
-      }
-
-      if (!collectionWatches.watchedCollections().contains(coll)) {
-        // This collection is no longer interesting, stop watching.
-        log.debug("Uninteresting collection {}", coll);
-        return;
-      }
-
-      Set<String> liveNodes = ZkStateReader.this.liveNodes;
-      if (log.isInfoEnabled()) {
-        log.info(
-            "A cluster state change: [{}] for collection [{}] has occurred - updating... (live nodes size: [{}])",
-            event,
-            coll,
-            liveNodes.size());
-      }
-
-      refreshAndWatch(event.getType());
-    }
+    public void process(WatchedEvent event) {}
```

Comment:
Perhaps we should move https://github.com/cowpaths/fullstory-solr/pull/105/files#diff-9b1918d641f78a8c245306b47ffd7a4554a12771ed4cb42293a411537fe7b54dR396-R510 to here.

It seems more appropriate for watcher to handle watched events instead of watch itself doing it. It confusing that this watcher to do nothing and the actual watch handling is defined elsewhere